go/cmd/abalone_example_ai/main.go

```diff
@@ -0,0 +1,20 @@
+package main
+
+import "github.com/danmane/abalone/go/abalone"
+
+func main() {
+	agent := abalone.AgentInfo{
+		Owner: "btc",
+		Taunts: []string{
+			"U MAD BRO?",
+			"If you took an IQ test, the results would be negative.",
+			"I don't know what makes you so dumb but it really works",
+			"If brains were taxed, you'd get a rebate.",
+			"Zombies eat brains. You’re safe.",
+		},
+	}
+	abalone.Play(agent, func(s abalone.GameState) abalone.GameState {
```

Comment:
@danmane @ali01 refactored the ai example so authors can implement agents without needing to know any of the details about the HTTP transport.

The one thing this function signature lacks is context about the game being played. We may want to pass information about the current opponent.